sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -258,21 +277,59 @@ export class ApplicationOfferingChangeRequestInstitutionsController extends Base
    * @param payload information to create the new request.
    * @returns newly change request id created.
    */
+  @ApiNotFoundResponse({
+    description: "Application not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Study period overlap or offering program year mismatch or offering intensity mismatch.",
+  })
+  @ApiUnauthorizedResponse({
+    description: "The location does not have access to the offering.",
+  })
   @Post()
   async createApplicationOfferingChangeRequest(
     @UserToken() userToken: IInstitutionUserToken,
     @Param("locationId", ParseIntPipe) locationId: number,
     @Body() payload: CreateApplicationOfferingChangeRequestAPIInDTO,
   ): Promise<PrimaryIdentifierAPIOutDTO> {
-    // TODO: Apply the same validations from PIR.
-    const applicationOfferingChangeRequest =
-      await this.applicationOfferingChangeRequestService.createRequest(
-        locationId,
+    try {
+      await this.applicationService.applicationOfferingValidation(
         payload.applicationId,
         payload.offeringId,
-        payload.reason,
-        userToken.userId,
+        locationId,
       );
-    return { id: applicationOfferingChangeRequest.id };
+      const applicationOfferingChangeRequest =
+        await this.applicationOfferingChangeRequestService.createRequest(
+          locationId,
+          payload.applicationId,
+          payload.offeringId,
+          payload.reason,
+          userToken.userId,
+        );
+      return { id: applicationOfferingChangeRequest.id };
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case APPLICATION_NOT_FOUND:
+            throw new NotFoundException(
+              new ApiProcessError(error.message, error.name),
+            );
+          case STUDY_DATE_OVERLAP_ERROR:
+          case OFFERING_PROGRAM_YEAR_MISMATCH:
+          case OFFERING_INTENSITY_MISMATCH:
+            throw new UnprocessableEntityException(
+              new ApiProcessError(error.message, error.name),
+            );
+          case OFFERING_DOES_NOT_BELONG_TO_LOCATION:
+            throw new UnauthorizedException(
+              new ApiProcessError(error.message, error.name),
+            );
+        }
+      }
+      throw new InternalServerErrorException(
```

Comment:
We need to have the `error` logged here otherwise we will never know its content.